[BUG] Cannot delete dataset processing

qualif-datasets-1  | Caused by: java.lang.NullPointerException: Cannot invoke "org.shanoir.ng.datasetacquisition.model.DatasetAcquisition.getExamination()" because the return value of "org.shanoir.ng.dataset.model.Dataset.getDatasetAcquisition()" is null
qualif-datasets-1  | 	at org.shanoir.ng.dataset.controler.DatasetApiController.deleteDataset(DatasetApiController.java:151)
qualif-datasets-1  | 	... 115 common frames omitted

Todo: check parent (and grand parent recursively enventually ?)

--- a/shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java
+++ b/shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/DatasetApiController.java
@@ -148,7 +148,16 @@ public class DatasetApiController implements DatasetApi {
 	public ResponseEntity<Void> deleteDataset(
 			final Long datasetId) throws EntityNotFoundException, RestServiceException {
 		try {
-			Long studyId = datasetService.findById(datasetId).getDatasetAcquisition().getExamination().getStudyId();
+			Dataset ds = datasetService.findById(datasetId);
+			if (ds == null) {
+				return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+			}
+			Long studyId;
+			if (ds.getDatasetProcessing() != null) {
+				studyId = ds.getDatasetProcessing().getStudyId();
+			} else {
+				studyId = ds.getDatasetAcquisition().getExamination().getStudyId();
+			}
 
 			datasetService.deleteById(datasetId);
 			solrService.deleteFromIndex(datasetId);

--- a/shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/service/DatasetProcessingServiceImpl.java
+++ b/shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/service/DatasetProcessingServiceImpl.java
@@ -25,6 +25,7 @@ import org.shanoir.ng.shared.exception.RestServiceException;
 import org.shanoir.ng.shared.exception.ShanoirException;
 import org.shanoir.ng.solr.service.SolrService;
 import org.shanoir.ng.utils.Utils;
+import org.shanoir.ng.vip.resource.ProcessingResourceService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
@@ -48,6 +49,9 @@ public class DatasetProcessingServiceImpl implements DatasetProcessingService {
 
     @Autowired
     private ProcessedDatasetService processedDatasetService;
+
+    @Autowired
+    ProcessingResourceService processingResourceService;
 
     @Autowired
     private DatasetService datasetService;
@@ -99,6 +103,9 @@ public class DatasetProcessingServiceImpl implements DatasetProcessingService {
     public void deleteById(final Long id) throws ShanoirException, RestServiceException, SolrServerException, IOException {
         final Optional<DatasetProcessing> entity = repository.findById(id);
         entity.orElseThrow(() -> new EntityNotFoundException("Cannot find dataset processing [" + id + "]"));
+
+        // delete associated ressources
+        processingResourceService.deleteByProcessingId(id);
 
         for (Dataset ds : entity.get().getOutputDatasets()) {
             datasetService.deleteById(ds.getId());

